### PR TITLE
add component-lifecyle content

### PR DESCRIPTION
**Pull request by joelhawksley**
This PR introduces a `Lifecycle` page to the design guidelines on `primer.style`.

## Context

As part of our goals we're working on increasing the maturity of Primer, to make it more robust and consistent, and to create clarity for people working with the system. Our component libraries have diverged as we experimented and developed different parts of Primer, and we don't have all the pattern coverage we think consumers need across the system. These criteria are meant to establish baseline expectations for components in Primer as they mature over time.

## Notes

We had originally titled these levels `maturity statuses` but I feel that the term `Lifecycle` better captures the intention for this to be a process that we move components through.  I also used the term `milestone` in place of `status` for similar reasons.

### CODE REVIEW

File: content/component-lifecycle.mdx [hunk 1]
```diff
@@ -2,5 +2,52 @@
...
+
+## Beta
+
+The component meets most maturity criteria, and use is encouraged for most scenarios.
+
+- <details><summary>The component is used in a production application in multiple instances.</summary>Expectation is three or more instances in dotcom for Primer ViewComponents, at least one or more instances for Primer React. Uses of the component have been reviewed for correctness.</details>
+- <details><summary>Usage guidelines and configuration documentation covers common use cases.</summary>Implementation of additional features (secondary priorities) is expected during beta. Additional and most common use cases should be documented by this phase. Documentation should include a sandbox environment such as Storybook.</details>
+- <details><summary>The design has been reviewed and any resulting issues have been addressed.</summary>Systems Designers have reviewed the current implementation and how it’s used in production.</details>
+- The accessibility team has manually reviewed the component and any outstanding issues have been fixed.
+- <details><summary>The component does not introduce unnecessary performance regressions to the user experience or the Primer library.
```
Comment (colebemis) [inline content/component-lifecycle.mdx:31]:
```suggestion
- The component does not introduce unnecessary performance regressions to the user experience or the Primer library.
```

File: content/component-lifecycle.mdx [hunk 2]
```diff
@@ -2,5 +2,52 @@
...
+Proof-of-concept built outside of Primer.
+
+- The component exists as a proof-of-concept built outside of Primer, often in a GitHub application such as GitHub.com.
+
+## Alpha
+
+The component is ready for preliminary usage, with breaking changes expected.
+
+- The component does not have external dependencies. This could be external libraries or dependencies on libraries in an application (like GitHub.com)
+- The component is compatible with color modes and can adapt to different themes. The component does not reference any hard-coded static values and uses functional variables. Any temporary or app-level variables have been removed. Theme values (such as a new brand scheme) can be swapped out without touching the public API of the component.
```
Comment (vdepizzol) [inline content/component-lifecycle.mdx:18]: A little late to the party, but I think we should make sure components are responsive friendly before hitting a `beta` status. Something along these lines:

> - "The component is designed with responsiveness in mind. The component can adapt to different screen sizes, from mobile to large desktop screens. The component interactive areas must be friendly to the touch on devices with a coarse pointer method.

Under **Beta**, we could also update include the following paragraph to mention responsiveness reviews:

> The design has been reviewed and any resulting issues have been addressed. Systems Designers have reviewed the current implementation, **validated its responsive strategy**, and how it’s used in production.

@joelhawksley @yaili @colebemis let me know your thoughts, I can open a new PR with this addendum!

Comment (joelhawksley) [reply content/component-lifecycle.mdx:18]: @vdepizzol that generally sounds good to me!

Comment (vdepizzol) [reply content/component-lifecycle.mdx:18]: Opened a PR here: https://github.com/primer/contribute/pull/2/files

**Pull request merged**